### ISSUE TITLE
Labels or Instructions of form fields (WCAG 3.3.2)

### ISSUE THREAD

**Issue by VivienDZB** [slidewiki/slidewiki-platform#1243]
**Request:** The intent of this criterion is to have content authors place instructions or labels that identify the controls in a form so that users know what input data is expected.

**Issues:** 
1. https://slidewiki.org/
- The form fields in the registration dialog are not adequately labeled. (**This also effects other pages!**)

2. https://slidewiki.org/search?keywords=accessibility&sort=score 
- The form field "Type your search terms here" only has a placeholder text. This is not sufficient, it is not reliably reproduced by screen readers.

3. https://slidewiki.org/deck/127411/_/slide/836159-2/836159-2:1/edit ; https://slidewiki.org/deck/127411/_/slide/836159-2/836159-2:1/markdownEdit

- The text fields on the page for editing the slides are input fields that do not contain an `aria-label` element.

4. https://slidewiki.org/search?keywords=accessibility&sort=score
- The input field for the search, as well as some fields within the "Advanced options" do not have a `label` element.

**Solution:**
Form fields should be labeled using the elements `label` or `legend`.
